src/rdkafka_fetcher.c

```diff
@@ -673,7 +836,11 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
         }
 
         /* Top level tags */
-        rd_kafka_buf_skip_tags(rkbuf);
+        rd_kafka_buf_read_tags(rkbuf, rd_kafkap_Fetch_reply_tags_parse,
+                               &FetchTags);
+
+        rd_kafka_handle_Fetch_metadata_update(rkb, &FetchTags);
+        rd_kafkap_Fetch_reply_tags_destroy(&FetchTags);
```

Comment:
These two lines should go just before the return because `rd_kafka_buf_parse_fail` can goto `err_parse` and `rd_kafkap_Fetch_reply_tags_destroy` be called twice